pkg/capabilities/consensus/ocr3/requests/handler.go

```diff
@@ -0,0 +1,146 @@
+package requests
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+type responseCacheEntry struct {
+	response  *Response
+	entryTime time.Time
+}
+
+type Handler struct {
+	services.StateMachine
+
+	lggr logger.Logger
+
+	mux             sync.Mutex
+	store           *Store
+	responseCache   map[string]*responseCacheEntry
+	cacheExpiryTime time.Duration
+
+	stopCh services.StopChan
+	wg     sync.WaitGroup
+}
+
+func NewHandler(lggr logger.Logger, s *Store, responseExpiryTime time.Duration) *Handler {
+	return &Handler{
+		lggr:            lggr,
+		store:           s,
+		responseCache:   map[string]*responseCacheEntry{},
+		cacheExpiryTime: responseExpiryTime,
+		stopCh:          make(services.StopChan),
+	}
+}
+
+func (h *Handler) SendResponse(resp *Response) {
+	req := h.getRequestForResponse(resp)
+	if req != nil {
+		req.sendResponse(resp)
+	}
+}
+
+func (h *Handler) SendRequest(r *Request) error {
+	resp, err := h.addRequest(r)
+	if err != nil {
+		return fmt.Errorf("failed to add request: %w", err)
+	}
+
+	if resp != nil {
+		r.sendResponse(resp)
+		return nil
+	}
+
+	return nil
+}
+
+func (h *Handler) ResponseCacheSize() int {
+	h.mux.Lock()
+	defer h.mux.Unlock()
+
+	return len(h.responseCache)
+}
+
+func (h *Handler) Start(_ context.Context) error {
+	return h.StartOnce("RequestHandler", func() error {
+		h.wg.Add(1)
+		go func() {
+			defer h.wg.Done()
+			ctx, cancel := h.stopCh.NewCtx()
+			defer cancel()
```

Comment:
You can put these 3 lines inside checkForResponseExpiry() and just run "go checkForResponseExpiry()" instead of having another function around it

Reply:
:) same reason as other PR, I think it is better to colocate the code that coordinates the routines using wait groups in the Start and Close methods as it makes the threading used by the service clearer (IMO), its a preference I am not wedded to it